src/corona_hakab_model/manager.py

```diff
@@ -158,37 +162,111 @@ def step(self):
-        new_results = self.pending_test_results.advance()
-        for agent, test_result, _ in new_results:
-            if test_result:
-                if not self.ever_tested_positive_vector[agent.index]:
-                    # TODO: awful late night implementation, improve ASAP
-                    # set isolation date
-                    self.step_to_isolate_agent[agent.index] = self.current_step + self.consts.isolate_after_num_day
-                    self.new_detected_daily += 1
-
-            agent.set_test_result(test_result)
-
+    def progress_isolations(self):
+        detected_positive_now = self.tested_positive_vector & \
+                                (self.date_of_last_test == self.current_step)
+        detected_positive_to_isolate = detected_positive_now & (self.agents_in_isolation == IsolationTypes.NONE)
+        detected_positive_indices = np.nonzero(detected_positive_to_isolate)[0]
+        for agent_index in detected_positive_indices:
+            # Get number of days agent is isolated
+            if self.agents_in_isolation[agent_index] != IsolationTypes.NONE:
+                # TODO: Need to take into account when got out of isolation also
+                number_of_days_isolated = max(0,
+                                              self.current_step - self.step_to_isolate_agent[agent_index])
+            else:
+                number_of_days_isolated = 0
+            weekly_connection_isolation_ratio = 1 - min(number_of_days_isolated / 7, 1)
+            for connected_agents in self.connection_data.connected_ids_by_strength[agent_index].values():
+                for daily_agent_id in connected_agents.daily_connections:  # All of this need to be home isolated
+                    # If were not isolated, and not sick, needs to
+                    if self.agents_in_isolation[daily_agent_id] == IsolationTypes.NONE and \
+                            not self.tested_positive_vector[daily_agent_id]:
+                        if self.step_to_isolate_agent[daily_agent_id] < self.current_step:
+                            self.step_to_isolate_agent[daily_agent_id] = self.current_step + \
+                                                                         self.consts.isolate_after_num_day
+                agents_to_iterate = list(filter(lambda index: (self.agents_in_isolation[index] == IsolationTypes.NONE)
+                                                              and not self.tested_positive_vector[index],
+                                                connected_agents.weekly_connections))
+                non_sick_agents = self.medical_machine.default_state_upon_infection.agents
+                shuffle(agents_to_iterate)
+                agents_to_iterate.sort(key=lambda index: self.agents[index] in non_sick_agents, reverse=True)
+                how_many_to_isolate = round(len(agents_to_iterate) * weekly_connection_isolation_ratio)
+                j = 0
+                for i in range(how_many_to_isolate):
```

Comment:
How do you enforce that the first "first circle" that will be isolated are the *currently infected* agents?
Not necessarily the symptomatic ones, which should isolate themselves, but anyone that is sick.

Reply:
Read the new comment tell me if now it answers